### PR TITLE
ROX-22047: OCM integration form

### PR DESCRIPTION
**Pull request by ivan-degtiarenko**
## Description

Similar to https://github.com/stackrox/stackrox/pull/9831, adds an integration form for OCM as a cloud source.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Screenshot 2024-02-16 at 17 39 43
Screenshot 2024-02-16 at 17 40 02
Screenshot 2024-02-16 at 17 40 11

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/OcmIntegrationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    cloudSource: {
+        id: '',
+        name: '',
+        type: 'TYPE_OCM',
+        credentials: {
+            secret: '',
+        },
+        skipTestIntegration: true,
+        ocm: {
+            endpoint: '',
```
Comment (stehessel) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/OcmIntegrationForm.tsx:59]: Can you default to `https://api.openshift.com`?

File: ui/apps/platform/src/Containers/Integrations/utils/integrationsList.ts [hunk 2]
```diff
@@ -290,7 +290,13 @@ export const paladinCloudDescriptor: CloudSourceDescriptor = {
     type: 'paladinCloud',
 };
 
-const cloudSourceDescriptors = [paladinCloudDescriptor];
+export const ocmDescriptor: CloudSourceDescriptor = {
+    image: redhat,
+    label: 'OCM',
```
Comment (stehessel) [inline ui/apps/platform/src/Containers/Integrations/utils/integrationsList.ts:295]: Maybe we should write it out as `OpenShift Cluster Manager` for the card?

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/OcmIntegrationForm.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                            id="cloudSource.name"
+                            value={values.cloudSource.name}
+                            onChange={onChange}
+                            onBlur={handleBlur}
+                            isDisabled={!isEditable}
+                        />
+                    </FormLabelGroup>
+                    <FormLabelGroup
+                        isRequired
+                        label="OCM endpoint"
```
Comment (stehessel) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/OcmIntegrationForm.tsx:129]: I think the "OCM" is redundant here. But looking at other integrations, we have both variants already (e.g. "Registry endpoint" for Artifact and just "Endpoint" for ECR).

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/OcmIntegrationForm.tsx:129]: Unless it is ambiguous to primary audience for this integration, just **Endpoint** seems like a solid pattern.

**Review by pedrottimark — APPROVED** [review]

**Comment by kcarmichael08** [comment]
This is late, sorry! Spelling it out is correct - Red Hat OpenShift Cluster Manager, approved short form OpenShift Cluster Manager - see product naming guide: https://docs.google.com/spreadsheets/d/1DLS_lS3VKidgZIvcLmLp9BoiqptkvqHWfe1D5FD2kfk/edit#gid=1375785039

**Review by stehessel — APPROVED** [review]

**Pull request merged**